WeLoveMangaOne: fix chapter and page logic

vetleledaal opened the pull request:
Closes #17195 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

arkon commented on multisrc/overrides/fmreader/welovemangaone/src/WeLoveMangaOne.kt:
```diff
@@ -2,14 +2,39 @@ package eu.kanade.tachiyomi.extension.ja.welovemangaone
         GET("$baseUrl/manga-list.html?page=$page&sort=last_update")
 
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val mangaId = MID_URL_REGEX.find(manga.url)
+            ?.groupValues
+            ?.get(1)
+            ?: throw Exception("Could not find manga id")
+
+        val xhrUrl = "$baseUrl/app/manga/controllers/cont.Listchapter.php".toHttpUrl().newBuilder()
+            .addQueryParameter("mid", mangaId)
+            .build()
+
+        val xhrRequest = GET(xhrUrl.toString(), headers)
+        val xhrResponse = client.newCall(xhrRequest).execute()
+
+        val chapterElements = xhrResponse.asJsoup().select(chapterListSelector())
```
Couldn't this fit within a regular override of `chapterListRequest`?

arkon commented on multisrc/overrides/fmreader/welovemangaone/src/WeLoveMangaOne.kt:
```diff
@@ -41,4 +66,26 @@ class WeLoveMangaOne : FMReader("WeLoveMangaOne", "https://welovemanga.one", "ja
         return chapterDate.timeInMillis
     }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val chapterId = document.selectFirst("#chapter")
+            ?.`val`()
+            ?: throw Exception("Could not find chapter id")
+
+        val xhrUrl = "$baseUrl/app/manga/controllers/cont.listImg.php".toHttpUrl().newBuilder()
+            .addQueryParameter("cid", chapterId)
+            .build()
+
+        val xhrRequest = GET(xhrUrl.toString(), headers)
+        val xhrResponse = client.newCall(xhrRequest).execute()
+
+        val xhrDocument = xhrResponse.asJsoup()
```
Similarly for `pageListRequest`

vetleledaal commented:
Switched over:
| From               | To                   |
| ------------------ | -------------------- |
| `fetchChapterList(manga: SManga): Observable<List<SChapter>>` | `chapterListRequest(manga: SManga): Request` |
| `pageListParse(document: Document): List<Page>`    | `pageListRequest(chapter: SChapter): Request`    |

Also rebased on top of master, hence the force push
